### PR TITLE
New playbook for adding nodes to existing OSE environments

### PR DESCRIPTION
**Pull request by oybed**
#### What does this PR do?

Adding in playbook for how to add OpenShift nodes to an existing environment.
#### How should this be manually tested?

Follow the "quick start" instructions:
https://github.com/rhtconsulting/openshift-playbooks/#quickstart-guide

Afterwards navigate to the "Playbooks >> Operationalizing >> OpenShift Enterprise 3 Adding Nodes - ScaleUp" 
#### Is there a relevant Issue open for this?

N/A
#### Who would you like to review this?

/cc @etsauer @sabre1041

### CODE REVIEW

File: playbooks/operationalizing/add_node.adoc [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+== Prerequisites
+
+The following prerequisites are required prior to beginning the steps to "ScaleUP" the environment 
+
+* A healthy and operational OpenShift Environment
+** Version 3.1.1 is recommended, but earlier 3.1.x versions may work as well after updating the "openshift-ansible" package
+* The current "inventory" file used for install (or any previous scaleup runs) of the OpenShift Environment
+* Additional VMs / hosts to be used for the node(s)
+** Needs to be running the correct OS release and "aligned" with the rest of the nodes in the OpenShift Environment.
+
```
Comment (etsauer) [inline playbooks/operationalizing/add_node.adoc]: @oybed I assume that for these hosts, each one needs to go through the host prep steps? I think that last bullet needs to be more explicit about that.

File: playbooks/operationalizing/add_node.adoc [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+== Goals
+
+Add in new OpenShift node(s) for increased capacity 
+
+== OpenShift ScaleUP
+Before the scaleup can be performed on the new node(s), make sure to install and configure the VM/host to resemble the rest of the environment. This includes, but not limited to:
+
+* Configuring and enabling separate disk partitions (logs, openshift, docker, etc.)
```
Comment (etsauer) [inline playbooks/operationalizing/add_node.adoc]: Again, I would roll this up to the Pre-requisites, and include links to the host prep section of the docs.

File: playbooks/operationalizing/add_node.adoc (hunk 1 above)
Comment (oybed) [inline playbooks/operationalizing/add_node.adoc]: @etsauer ok, will add some clarifying text

File: playbooks/operationalizing/add_node.adoc (hunk 2 above)
Comment (oybed) [inline playbooks/operationalizing/add_node.adoc]: ok, will do

**Comment by oybed** [comment]
@etsauer PR has been updated according to your feedback/comments. Please re-check and merge if you think it's ready.

**Comment by JaredBurck** [comment]
@oybed @etsauer  Followed the "quick start" instructions to review addnode playbook. Created new OSE env in OS1 with 1 master + 2 nodes.  Followed addnode playbook to add two additional nodes to OSE cluster. There were a couple minor issues I experienced while testing and verifying this worked as expected. The docker service would fail on DNS host when running scaleup, which would cause ansible scaleup to fail. Maybe just an OS1 issue? Restarting the docker service on that host and rerunning the ansible scaleup playbook (either from beginning or retry) would successfully complete the scaleup. In the end, all validations passed and were successful.

Also found two Solutions articles on "how to add a node to OSE v3.0 [1] and v3.1 [2]".  Version 3.0 is a manual process, while 3.1 uses the ansible scaleup (though varies a bit from the playbook).

[1] https://access.redhat.com/solutions/1983683
[2] https://access.redhat.com/solutions/2150381

File: playbooks/operationalizing/add_node.adoc [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+Øystein Bedin <bedin@redhat.com>
+v1.0, 02-Feb-2016
+:scripts_repo: https://github.com/rhtconsulting/rhc-ose
+:toc: macro
+:toc-title:
+
+toc::[]
+
+After running OpenShift for awhile, there may be a need to add in additional capacity for hosting applications and pods. This can be done by adding additional "compute nodes", or OpenShift nodes, to the existing infrastructure without having to go through a re-install. One option is to re-run the installation with an updated inventory file, and additional nodes. However, this can be a bit risky as it will revert custom configurations made after initial installation back to "fresh install" state. A better mechanism is to use the "scaleup" playbook part of the "openshift-ansible" installation implementation. This document describes the steps necessary to do this work and gotcha's that you may have to consider. 
+
```
Comment (sabre1041) [inline playbooks/operationalizing/add_node.adoc]: and *add* additional nodes

File: playbooks/operationalizing/add_node.adoc [hunk 4]
```diff
@@ -0,0 +1,120 @@
+---
+---
+= OpenShift Enterprise 3 Adding nodes - "ScaleUP"
+Øystein Bedin <bedin@redhat.com>
+v1.0, 02-Feb-2016
+:scripts_repo: https://github.com/rhtconsulting/rhc-ose
+:toc: macro
+:toc-title:
+
+toc::[]
```
Comment (sabre1041) [inline playbooks/operationalizing/add_node.adoc]: Heading section does not render properly

File: playbooks/operationalizing/add_node.adoc [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+After running OpenShift for awhile, there may be a need to add in additional capacity for hosting applications and pods. This can be done by adding additional "compute nodes", or OpenShift nodes, to the existing infrastructure without having to go through a re-install. One option is to re-run the installation with an updated inventory file, and additional nodes. However, this can be a bit risky as it will revert custom configurations made after initial installation back to "fresh install" state. A better mechanism is to use the "scaleup" playbook part of the "openshift-ansible" installation implementation. This document describes the steps necessary to do this work and gotcha's that you may have to consider. 
+
+
+== Prerequisites
+
+The following prerequisites are required prior to beginning the steps to "ScaleUP" the environment 
+
+* A healthy and operational OpenShift Environment
+** Version 3.1.1 is recommended, but earlier 3.1.x versions may work as well after updating the "openshift-ansible" package
+* The current "inventory" file used for install (or any previous scaleup runs) of the OpenShift Environment
```
Comment (sabre1041) [inline playbooks/operationalizing/add_node.adoc]: This [inventory](http://docs.ansible.com/ansible/intro_inventory.html) link might be helpful to include

File: playbooks/operationalizing/add_node.adoc [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+<1> OpenShift version string for the environment
+<2> New OpenShift nodes to be added
+<3> New node listed in the [nodes] host group
+<4> Same as 3., new node listed in the [nodes] host group
+
+Next, run the ansible-playbook for scaleup with the inventory file above:
+
+    ansible-playbook -i <path_to_inventory_file> /usr/share/ansible/openshift-ansible/playbooks/byo/openshift-cluster/scaleup.yml
+
+== Post ScaleUP tasks
```
Comment (sabre1041) [inline playbooks/operationalizing/add_node.adoc]: A link to the [Verification Guide](https://github.com/rhtconsulting/openshift-playbooks/blob/master/playbooks/installation/install_validation.adoc) would also be helpful

**Comment by JaredBurck** [comment]
OpenShift docs now have an Adding Nodes to an Existing Cluster section in the Advanced Installation guide.  The process is very similar except how the inventory file is setup.  Should probably make these more consistent or explain why we do it differently. 

[1] https://docs.openshift.com/enterprise/3.1/install_config/install/advanced_install.html#adding-nodes-advanced

**Comment by JaredBurck** [comment]
closing PR as this has been added to the official docs and is no longer needed.

**Pull request closed without merging**